remote-api/src/main/java/org/alfresco/rest/api/impl/rules/RuleSetLoader.java

```diff
@@ -45,15 +53,32 @@
      * @param includes A list of fields to include.
      * @return The rule set object.
      */
-    protected RuleSet loadRuleSet(NodeRef ruleSetNodeRef, List<String> includes)
+    protected RuleSet loadRuleSet(NodeRef ruleSetNodeRef, NodeRef folderNodeRef, List<String> includes)
     {
         String ruleSetId = ruleSetNodeRef.getId();
         RuleSet ruleSet = RuleSet.of(ruleSetId);
 
-        if (includes != null && includes.contains("owningFolder"))
+        if (includes != null)
         {
             NodeRef parentRef = nodeService.getPrimaryParent(ruleSetNodeRef).getParentRef();
-            ruleSet.setOwningFolder(parentRef);
+            if (includes.contains(OWNING_FOLDER))
+            {
+                ruleSet.setOwningFolder(parentRef);
+            }
+            if (includes.contains(INCLUSION_TYPE))
+            {
+                if (parentRef.equals(folderNodeRef))
+                {
+                    ruleSet.setInclusionType(OWNED);
+                }
+                else
+                {
+                    boolean linked = nodeService.getParentAssocs(ruleSetNodeRef)
+                                           .stream().map(ChildAssociationRef::getParentRef)
+                                           .anyMatch(folderNodeRef::equals);
+                    ruleSet.setInclusionType(linked ? LINKED : INHERITED);

```

Comment:
I'm wondering if there exists a case when rule set could be linked and inherited in the same time. What should be the inclusion value for this case?

Reply:
The comment and extra unit tests are now added.

Reply:
Thanks for clarification and comment